parlai/tasks/taskmaster/agents.py

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+from parlai.core.teachers import FixedDialogTeacher
+from . import tm_utils
+import json
+
+
+class SelfDialogueTeacher(FixedDialogTeacher):
+    """
+    Teacher for written two-person dialogues with labels being responses for
+    the previous statement. The data is traversed twice (doubled), once for
+    modelling USER replies and once for modelling ASSISTANT replies.
+    """
+
+    def __init__(self, opt, shared=None):
+        super().__init__(opt)
+        opt['fn'] = "self-dialogs.json"
+
+        if 'exclude-invalid-data' not in opt:
```

Comment:
```python
@staticmethod
    def add_cmdline_args(argparser):
        agent = argparser.add_argument_group('Corrupt-Example-Arguments')
        agent.add_argument(
            '--exclude-invalid-data',
            type='bool',
            default=True,
            help='Whether to include corrupt examples in the data',
        )
```

Does this look like the way to do it?
I assumed this would be called automatically and I don't need to add anything apart from this.

Reply:
yes, that is the correct way to do it

Reply:
yes, that is the correct way to do it

Reply:
Great I'll add that. I figured this option is used only for woz so incorporating it just there!